### PR TITLE
feat: markdown support

### PR DESCRIPTION
**Pull request by philemonputra**
### Description
- Created markdown features such as bold, italics, clickable links,  tags, etc.

### Related Issues
- #31 

### Type of Change
- [x] New feature

### Testing Details
- Tested each feature of the markdown listed in src/utils.java

### Checklist
- [ ] Code follows the style guide.
- [x] I have performed a self-review of my code.
- [ ] Documentation has been updated (if necessary).
- [x] Tests have been added or updated.

### Sample:
image

### CODE REVIEW

**Comment by MitchelHsu** [comment]
I think we should assign reviewers, instead of assignees lol. Mind moving to the Reviewers section?
Screenshot 2024-11-25 at 10 25 32 AM

**Review by kenny-lfx — APPROVED** [review]

File: tests/unit/Utils.test.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
+import { parseMarkdown } from '../../src/js/utils';
+
+describe('Util ParseMarkdown', () => {
```
Comment (MitchelHsu) [inline tests/unit/Utils.test.js:3]: This is great @philemonputra @jet0612, thanks 🙏. I would also add a test that has multiple MD format text, and the function should also convert the ones that are in MD and should not convert the ones that are not,

Comment (philemonputra) [reply tests/unit/Utils.test.js:3]: Got it!

**Review by kenny-lfx — APPROVED** [review]

File: tests/unit/Utils.test.js (hunk 1 above)
Comment (philemonputra) [reply tests/unit/Utils.test.js:3]: Just added a test to accommodate multiple markdown features

**Review by kenny-lfx — APPROVED** [review]

**Review by Shyam-Sundar-Bharathi — APPROVED** [review]

**Review by ZihanLing — APPROVED** [review]

**Pull request merged**